editor/editor_node.cpp

```diff
@@ -238,185 +238,182 @@ void EditorNode::_unhandled_input(const Ref<InputEvent> &p_event) {
 
 void EditorNode::_notification(int p_what) {
 
-	if (p_what == NOTIFICATION_EXIT_TREE) {
-
-		editor_data.save_editor_external_data();
-		FileAccess::set_file_close_fail_notify_callback(NULL);
-		log->deinit(); // do not get messages anymore
-	}
-	if (p_what == NOTIFICATION_PROCESS) {
-
-		if (opening_prev && !confirmation->is_visible())
-			opening_prev = false;
+	switch (p_what) {
+		case NOTIFICATION_EXIT_TREE: {
```

Comment:
Could you move `EXIT_TREE` below `ENTER_TREE`? That would be a more natural order.